### DISCUSSION TITLE
Feature request: update user password requirements

### DISCUSSION THREAD

**Discussion by theMackabu** [pocketbase/pocketbase#356] [Q&A] [answered]
It might be a good idea to require the user to enter their current password to verify its them before updating email or passwords, even if they are already logged in.

**Comment by ganigeorgiev** [comment, marked as answer]
The email change confirmation request already requires the users to provide their existing password before the change is applied (see https://github.com/pocketbase/pocketbase/blob/master/forms/user_email_change_confirm.go).

For passwords - users are allowed to change their password only through the email forgotten/reset-password flow. If you need to provide your users another way to change their password, you could always create a custom endpoint.

**Reply by jimafisk** [reply]
@ganigeorgiev do you anticipate eventually adding a way to reset passwords simply by confirming the current password (not requiring email confirmation)? We're using PB as a frameworks so we can add a custom endpoint, just seems like this might be a common workflow that folks will want. Thanks!

**Reply by ganigeorgiev** [reply]
@jimafisk With v0.8+ it should be supported if you provide the `oldPassword` field. Admins and users with "Manage access" can change the account password directly.
You can find more documentation in the "Admin UI > API Preview > Update" or in the general https://pocketbase.io/docs/api-records/ > "Update record" section.

**Reply by jimafisk** [reply]
Oh perfect :tada:. Thanks for pointing me to it!

**Reply by jimafisk** [reply]
@ganigeorgiev I implemented the password change via the `oldPassword` field and it works great. I did want to note one small thing, if you submit an empty `password` it appears to give a `200` response. I assume this is because there is no update actually taking place. If you pass any value as `password` all the validation works as expected (matching passwords, password complexity, valid `oldPassword`, etc).

I was surprised that blank submission validation appeared to be missing. There might be a reason for this behavior, like it's needed for updating other fields when `password` is not changing. Would there be a way to check if the `password` key was passed in the request (i.e. someone intended to change it but didn't pass a value)? Not super high priority, I can work around it, just wanted to put it on your radar. Thanks!

**Reply by ganigeorgiev** [reply]
@jimafisk I'll update the validator triggers in the next release.

**Reply by jacksonsophat** [reply]
Has this function been updated yet?

**Reply by ganigeorgiev** [reply]
@jacksonsophat I'm not sure to what you are referring. Please file a new discussion with more details.

**Reply by ganigeorgiev** [reply]
@jacksonsophat If you mean the password validation triggers - kind of.
Depending on the scenario, the fields validation will be triggered if at least one of the `oldPassword`, `password` or `passwordConfirm` fields is set, aka. if only `oldPassword` is filled you should get `password` and `passwordConfirm` blank validations. If all of 3 fields are empty then the fields will be ignored. This is because there is no reliable way to detect whether the form was filled from a client request or programmatically, aka. we'll have to introduce some extra flags and/or introduce breaking changes converting the password fields to pointers and at least for now I don't think it is worth doing it.
